homework02/allan/string.h

```diff
@@ -0,0 +1,41 @@
+
+#pragma once
+
+#include <iosfwd>
+#include <memory>
+
+class String
+{
+public:
+    String();
+    String(const char *);
+    String(const String& other);
+                String(const char *, int capactiy);
+    String& operator=(String other);
```

Comment:
Why not passing by const ref here?

Reply:
I needed a copy anyway in the implementation. This is a handy way of doing it.